src/ralph/deployment/deployment.py

```diff
@@ -438,6 +439,92 @@ def check_ip_from_defined_network(address):
 
 
 @deployment_action(
+    verbose_name=_('Assign new IP'),
+    form_fields={
+        'ip_or_network': {
+            'field': ChoiceFieldWithOtherOption(
+                label=_('IP Address'),
+                other_field=forms.GenericIPAddressField(),
+                auto_other_choice=False,
+            ),
+            'choices': next_free_ip_choices,
+            'exclude_from_history': True,
+            'validation': validate_ip_address,
+        },
+    },
+    run_after=['assign_new_hostname'],
+)
+def assign_new_ip(cls, instances, ip_or_network, **kwargs):
+    for instance in instances:
+        if ip_or_network['value'] == OTHER:
+            ip_address = ip_or_network[OTHER]
+            ip = IPAddress.objects.create(address=ip_address)
+        else:
+            network = Network.objects.get(
+                pk=ip_or_network['value']
+            )
+            ip = network.issue_next_free_ip()
+        logger.info('Assigning {} to {}'.format(ip, instance))
+        ethernet = Ethernet.objects.create(base_object=instance)
+        logger.info('Bounding {} to {} ethernet'.format(ip, ethernet))
+        ip.ethernet = ethernet
+
+        hostname = kwargs['shared_params']['hostnames'].get(instance.pk)
+        if hostname:
+            ip.hostname = hostname
+
+        ip.save()
+
+
+def base_object_ip_choices(actions, objects):
+    ipaddresses = []
+    for instance in objects:
+        for ip in instance.ipaddresses:
+            ipaddresses.append(
+                [ip.id, '{} - {}'.format(ip.address, ip.hostname)]
+            )
+    return ipaddresses
+
+
+def base_object_network_choices(actions, objects):
+    networks = []
+    for instance in objects:
+        for network in instance._get_available_networks():
+            networks.append(
+                [network.id, network]
+            )
+    return networks
+
+
+@deployment_action(
+    verbose_name=_('Replace IP'),
+    form_fields={
+        'ipaddress': {
+            'field': ChoiceFieldWithOtherOption(
```

Comment:
it should be basic choice field (we don't allow other value here - we should choose one of current IPs of this base object)